src/query/sensor/processors/sensor.processor.ts

```diff
@@ -131,46 +140,52 @@ export class SensorProcessor {
       sensorData = {...sensorData, typeDetails: event.typeDetails};
     }
 
-    Sensor.updateOne({_id: event.sensorId}, sensorData, (err) => {
-      if (err) {
-        this.logError(event);
-      }
-    });
+    const sensor: ISensor = await Sensor.findByIdAndUpdate(
+      event.sensorId,
+      sensorData,
+      this.errorCallback,
+    ).exec();
+
+    return sensor;
   }
 
-  async processDeleted(event: SensorDeleted): Promise<void> {
-    Sensor.deleteOne({_id: event.sensorId}, (err) => {
-      if (err) {
-        this.logger.error('Error while deleting projection.');
-      }
-    });
+  async processDeleted(event: SensorDeleted): Promise<ISensor> {
+    const sensor: ISensor = await Sensor.findByIdAndDelete(
+      event.sensorId,
+    ).exec();
+
+    return sensor;
```

Comment:
Tja ... dit is ook grappig! Een delete geeft een object terug dat niet meer bestaat 😄 Maar ja, wat anders?

Reply:
In gedachten leven de sensoren voort 😉 Kan natuurlijk alleen de object id teruggeven van de verwijderde sensor, maar nu is de return signature van elke process* functie Promise<ISensor> en daar wordt ik gewoon heel blij van, haha